### ISSUE TITLE
Even with LIBEV_EMBED=1 monkey_test test_threading.py fails

### ISSUE THREAD

**Issue by mcepl** [gevent/gevent#1502]
* gevent version: 1.5a3
* Python version: Please be as specific as possible: "cPython 3.7.3 from openSUSE/Tumbleweed package"
* Operating System: openSUSE/Tumbleweed, x86_64

### Description:

Running the test suite while building gevent package for openSUSE (and yes, against my better self, I have LIBEV_EMBED set to 1, see #1501)

```
[  317s]   ...................ssssssssssss.....................ss..........s.....................s.Fatal Python error: Py_EndInterpreter: not the last thread
[  317s]   
[  317s]   Current thread 0x00007f36aa4b86c0 (most recent call first):
[  317s]     File "/usr/lib64/python3.7/test/support/__init__.py", line 2709 in run_in_subinterp
[  317s]     File "/tmp/test_threadings3nfkgam.py", line 1004 in test_threads_join_2
[  317s]     File "/usr/lib64/python3.7/unittest/case.py", line 615 in run
[  317s]     File "/usr/lib64/python3.7/unittest/case.py", line 663 in __call__
[  317s]     File "/usr/lib64/python3.7/unittest/suite.py", line 122 in run
[  317s]     File "/usr/lib64/python3.7/unittest/suite.py", line 84 in __call__
[  317s]     File "/usr/lib64/python3.7/unittest/suite.py", line 122 in run
[  317s]     File "/usr/lib64/python3.7/unittest/suite.py", line 84 in __call__
[  317s]     File "/usr/lib64/python3.7/unittest/runner.py", line 176 in run
[  317s]     File "/usr/lib64/python3.7/unittest/main.py", line 271 in runTests
[  317s]     File "/usr/lib64/python3.7/unittest/main.py", line 101 in __init__
[  317s]     File "/tmp/test_threadings3nfkgam.py", line 1299 in <module>
[  317s]     File "/home/abuild/rpmbuild/BUILDROOT/python-gevent-1.5.0~a3-74.2.x86_64/usr/lib64/python3.7/site-packages/gevent/testing/monkey_test.py", line 86 in <module>
[  317s]     File "/usr/lib64/python3.7/runpy.py", line 85 in _run_code
[  317s]     File "/usr/lib64/python3.7/runpy.py", line 193 in _run_module_as_main
[  317s]
```

### What I've run:
```
cat <<'EOF' >> network_tests.txt
test__all__.py
test___config.py
test__doctests.py
test__examples.py
test__execmodules.py
test__getaddrinfo_import.py
test_httplib.py
test__socket_dns.py
test_socket.py
test__socket_ssl.py
test__ssl.py
test_ssl.py
test_threading.py
test_urllib2_localnet.py
test_urllib2net.py
test_wsgiref.py
EOF
export GEVENT_RESOLVER=thread
export TRAVIS=1
export APPVEYOR=1
export LANG=en_US.UTF-8
PYTHONPATH=%{buildroot}%{$python_sitearch} python3 -m gevent.tests --ignore network_tests.txt
```
[Complete build log with all details](https://github.com/gevent/gevent/files/4020312/_log_gevent.txt)

**Comment by jamadden** [comment]
I've done some looking, and I think you saw [bpo-36402](https://bugs.python.org/issue36402). In short, `test_threads_join_2` has long had some race conditions, and this failure mode is one of the ways that manifests. A re-run frequently succeeds.

The fix landed in Python 3.7.5.

**Comment by jamadden** [comment]
Closing since it seems resolved.
